coreutils_core/src/mktemp.rs

```diff
@@ -0,0 +1,88 @@
+//! Module with wrappers for libc mkstemp(3), mkdtemp(3)
+
+use std::{
+    fmt::{self, Display},
+    fs::File,
+    io::Error,
+    os::unix::io::FromRawFd,
+};
+
+#[derive(Debug, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct MktempError {
+    err: String,
+}
+
+/// A struct that represents a mktemp(3) result.
+/// This includes the file created, and the path to that file.
+#[derive(Debug)]
+pub struct Mktemp {
+    pub file: File,
+    pub path: String,
```

Comment:
I thing it could be better to use `BString` type, defined on `bstr` crate, since it may raise error due system info not always being UTF-8

Reply:
That should never happen, since the function creating the `Mktemp` instance is passed a string, and there's no libc that would replace the `X`s in the template with anything but ASCII letters and digits.